File: components/string-utilities/src/main/java/org/cloudfoundry/credhub/utils/VersionProvider.java
package org.cloudfoundry.credhub.utils;

import java.io.IOException;

import org.springframework.stereotype.Component;

@Component
public class VersionProvider {
  private String version;

  public VersionProvider(final ResourceReader resources) {
    super();
    try {
      version = resources.readFileToString("version").trim();
    } catch (final IOException | IllegalArgumentException e) {
      version = "0.0.0";
    }
  }

  public String currentVersion() {
    return version;
  }
}

File: components/string-utilities/src/main/kotlin/org/cloudfoundry/credhub/utils/VersionProvider.kt
package org.cloudfoundry.credhub.utils

import org.springframework.stereotype.Component
import java.io.IOException

@Component
class VersionProvider(
    resources: ResourceReader,
) {
    final var version: String =
        try {
            resources.readFileToString("version").trim { it <= ' ' }
        } catch (e: IOException) {
            "0.0.0"
        } catch (e: IllegalArgumentException) {
            "0.0.0"
        }

    fun currentVersion(): String = version
}
